runtime/lua/vim/iter.lua

```diff
@@ -592,6 +592,27 @@ function ListIter.rfind(self, f) -- luacheck: no unused args
   self._head = self._tail
 end
 
+--- Consumes an iterator, yielding only the first n values.
+---
+--- Example:
+---
+--- ```lua
+--- local it = vim.iter({ 1, 2, 3, 4 }):take(2)
+--- it:next()
+--- -- 1
+--- it:next()
+--- -- 2
+--- it:next()
+--- -- nil
+--- ```
+function Iter.take(self, n)
+  local i = 0
+  return self:filter(function()
+    i = i + 1
+    return i <= n
```

Comment:
wouldn't this be really inefficient if you take one element from a huge iterator?
Not really familiar with the iterators so can't tell if there is a better way

Reply:
Yeah, you're right. I think the use case would mostly be with smaller iterators or places where a limit is needed, but maybe it could be made to not consume the entire iterator?

Reply:
Couldn't the same argument be made for `skip` / `nth`, etc.?

Reply:
well I don't know how they are implemented but if they are the same way (using `filter`) then yes

Reply:
I think it could be made better for ListIters, let me see if I can get something working that doesn't consume the entire iterator.

Reply:
and `nth` is implemented with `skip`, and skip is just 

```lua
for _ = 1, n do
  local _ = self:next()
end
return self
```